addons/mail/static/src/discuss/core/web/discuss_core_web_service.js

```diff
@@ -24,24 +25,26 @@ export class DiscussCoreWeb {
 
     setup() {
         this.messagingService.isReady.then((data) => {
-            for (const channelData of data.channels) {
-                const thread = this.discussCoreCommonService.createChannelThread(channelData);
-                if (
-                    channelData.is_minimized &&
-                    channelData.state !== "closed" &&
-                    !this.ui.isSmall
-                ) {
-                    this.store.ChatWindow.insert({
-                        autofocus: 0,
-                        folded: channelData.state === "folded",
-                        thread,
-                    });
+            Record.MAKE_UPDATE(() => {
+                for (const channelData of data.channels) {
+                    const thread = this.discussCoreCommonService.createChannelThread(channelData);
+                    if (
+                        channelData.is_minimized &&
+                        channelData.state !== "closed" &&
+                        !this.ui.isSmall
+                    ) {
+                        this.store.ChatWindow.insert({
+                            autofocus: 0,
+                            folded: channelData.state === "folded",
+                            thread,
+                        });
+                    }
                 }
-            }
-            this.store.discuss.channels.isOpen =
-                data.current_user_settings.is_discuss_sidebar_category_channel_open;
-            this.store.discuss.chats.isOpen =
-                data.current_user_settings.is_discuss_sidebar_category_chat_open;
+                this.store.discuss.channels.isOpen =
+                    data.current_user_settings.is_discuss_sidebar_category_channel_open;
+                this.store.discuss.chats.isOpen =
+                    data.current_user_settings.is_discuss_sidebar_category_chat_open;
+            });
```

Comment:
should split `MAKE_UPDATE` to make it as small as possible to avoid missing unwanted side effect

in particular here, I'd suggest that only the `createChannelThread` should be batched

Reply:
because if `this.store.ChatWindow.insert` depends on (computed) thread data, they will miss the computed values (if they are computed at the end of the make_update)